### PR TITLE
refactor(jobs): make it generic and notify errors

### LINKED ISSUE #112: Put job in error when it fails
Job should be marked in error in database when it fails 3 times.

### PR DESCRIPTION
**Pull request by neoziro**
Closes #112

### CODE REVIEW

File: src/server/routes/configurePassport.js [hunk 1]
```diff
@@ -33,6 +31,7 @@ export default (passport) => {
           .insert({
             githubId: Number(profile.id),
             accessToken,
+            login: profile.username,
```
Comment (oliviertassinari) [inline src/server/routes/configurePassport.js]: Why not doing it in `getDataFromProfile`?

File: src/server/models/Build.test.js [hunk 2]
```diff
@@ -7,6 +7,7 @@ const baseData = {
   repositoryId: '1',
   baseScreenshotBucketId: '1',
   compareScreenshotBucketId: '2',
+  jobStatus: 'pending',
```
Comment (oliviertassinari) [inline src/server/models/Build.test.js]: Why, I don't think we need it.

File: src/modules/build/computeScreenshotDiff.test.js [hunk 3]
```diff
@@ -56,13 +56,15 @@ describe('computeScreenshotDiff', () => {
...
+    await computeScreenshotDiff(screenshotDiff, {
       s3,
       bucket: 'argos-screenshots-sandbox',
     })
-    expect(resultScreenshotDiff.score > 0).toBeTruthy()
-    expect(resultScreenshotDiff.jobStatus).toBe('complete')
-    expect(resultScreenshotDiff.s3Id).not.toBeUndefined()
+
+    await screenshotDiff.reload()
+    expect(screenshotDiff.score > 0).toBeTruthy()
```
Comment (oliviertassinari) [inline src/modules/build/computeScreenshotDiff.test.js]: true

File: src/server/models/Build.test.js (hunk 2 above)
Comment (neoziro) [inline src/server/models/Build.test.js]: We need it because your use `Build.fromJson`

**Pull request merged**